src/core/core.layoutService.js

```diff
@@ -79,6 +113,14 @@ module.exports = function(Chart) {
 				return (a.options.fullWidth ? 1 : 0) - (b.options.fullWidth ? 1 : 0);
 			});
 
+			// Ensure that boxes are sorted by order. Higher order means further away from the chart
+			leftBoxes.sort(function(a, b) {
```

Comment:
I think so, maybe `options.fullWidth` should not be used anymore to control the box order.